fuel-bft/src/metadata.rs

```diff
@@ -0,0 +1,210 @@
+use crate::{Error, HeightRound, Message, Network, RoundValidator, State};
+
+use fuel_crypto::borrown::Borrown;
+use fuel_crypto::{Keystore, PublicKey, Signer};
+
+use core::borrow::Borrow;
+
+/// Metadata required for the consensus state machine.
+pub trait Metadata {
+    /// Concrete error implementation
+    type Error: From<Error>
+        + From<fuel_crypto::Error>
+        + From<<<Self::Network as Network>::Message as Message>::Error>
+        + From<<<<Self::Network as Network>::Message as Message>::Signer as Signer>::Error>;
+
+    /// Set of validators for a given round
+    type Filtered: Iterator<Item = Self::PublicKey>;
+
+    /// Network implementation responsible for message broadcasting.
+    type Network: Network;
+
+    /// Borrowed public key for iterators of validators
+    type PublicKey: Send + Borrow<PublicKey>;
+
+    /// Sorted set of validators for a given round
+    type Sorted: Iterator<Item = Self::PublicKey>;
+
+    /// Check if the node is enabled to participate in the height
+    fn is_valid_for(&self, height: u64) -> bool;
+
+    /// Id of the key used for the round
+    fn id(
+        &self,
+        round: &HeightRound,
+    ) -> Result<&<<<<Self::Network as Network>::Message as Message>::Signer as Signer>::Keystore as Keystore>::KeyId, Self::Error>;
+
+    /// Validators filtered per height
+    fn filter_round(&self, height: u64) -> Result<Self::Filtered, Self::Error>;
```

Comment:
filter_round sounds kind of vague and makes other parts of the code harder for me to read. Maybe something like this?

```suggestion
    fn validators_at_round(&self, height: u64) -> Result<Self::Filtered, Self::Error>;
```

Reply:
https://github.com/FuelLabs/fuel-bft/pull/9/commits/5a0b1205568fdb404351ebc6636d2192c2f7cd08